### PR TITLE
Medieval mod inclusion+

### LINKED ISSUE #27092: Mod Inclusion Project: Medieval and Historic Content
Use the following issue as a space to discuss the inclusion of:

Mod Ident: Medieval_Stuff
Name: Medieval and Historic Content

Your comment should include:

### Suggestions for induction
(what parts of the mod should be included, can be as simple as just an item, a location or the complete inclusion)

### Incompatibilities
(what parts of the mod should not be included, either some objects / items or entire mod is conflicting with DDA theme [ for whatever reason ])

### PR DESCRIPTION
**Pull request by Mooses2k**
#### Summary
```SUMMARY: Content "Mod Inclusion - Medieval with tuning"```

#### Purpose of change
Fixes #27092

#### Describe the solution
Integrated the mod items and recipes and itemgroups and martial art into mainline. Did a little organization while I was at it, including separating museum stuff into its own itemgroup json (hope that's okay).

Improved the Medieval Swordsmanship martial art using my HEMA knowledge. Added a MORDHAU which was worth the price of admission for this PR.

Added remaining missing inferior items. Added more appropriate weapons available to Medieval Swordsmanship and, as per Kevin's recommendation, made it just for swordfighting, since spears etc use different techniques.

#### Additional context
[Some flavor to get you in the mood for this PR](https://www.youtube.com/watch?v=nFpoDlH6Sxs)

Thanks to Chaosvolt for making this mod.

### CODE REVIEW

**Comment by Rail-Runner** [comment]
> Let me know what I should do with the dusksword

I'd refluff it as a sword built using sci-fi materials like superalloy or whatever, made for sword collectors for the most part (would have to be extremely rare obviously). If this isn't good to have in mainline, move it to Bright Nights (probably fine to do this because it didn't spawn at all before).

**Comment by kevingranade** [comment]
> Let me know what I should do with the dusksword...is there a flag for fantasy stuff?

Either leave it in the mod or delete it, it looks like it never spawned in the first place, and it's definitely not ok for mainline.

**Comment by SirPendrak** [comment]
Martial Arts starting traits now can give you Med. Swordsmanship, but its descrieption still mentions only the standard 5 styles to choose from.

**Comment by Mooses2k** [comment]
Will delete the dusksword and alter the description of the martial art.

**Comment by Mooses2k** [comment]
A question has come up about how to handle shields.

Two thoughts occur to me:
1) FLAG on the item that increases time-to-aim against the bearer
2) For shields only, also add item's coverage rating to torso for the purposes of getting hit at range.

**Comment by kevingranade** [comment]
I prefer 3 for now, that deserves it's own discussion.
I'm leaning toward no new shield features in mainline until we get offhand wielding.

File: data/json/items/tools.json [hunk 1]
```diff
@@ -9061,6 +9061,43 @@
     "qualities": [ [ "CUT", 1 ], [ "BUTCHER", 8 ] ],
     "flags": [ "SHEATH_SWORD", "DURABLE_MELEE" ]
   },
+  {
+    "id": "khopesh",
+    "type": "TOOL",
+    "category": "tools",
+    "name": "pallet of dry adobe bricks",
```
Comment (anothersimulacrum) [inline data/json/items/tools.json]: Something is funky with this commit

**Comment by Mooses2k** [comment]
I need help with the tools,json thing. Rebasing and merging is confusing for me and I screwed something up with that file. Help?

**Comment by Mooses2k** [comment]
I think Travis needs to rerun. Says 1 failed but shows nothing failed?

**Comment by AMurkin** [comment]
There was an assertion error in `monster_test.cpp`.

File: data/json/items/melee.json [hunk 2]
```diff
@@ -2283,5 +2302,59 @@
...
+    "cutting": 28,
+    "material": [ "bronze", "wood" ],
+    "symbol": "/",
+    "color": "yellow",
+    "techniques": [ "WBLOCK_1", "IMPALE" ],
+    "qualities": [ [ "COOK", 1 ] ],
+    "flags": [ "SPEAR", "REACH_ATTACK", "DURABLE_MELEE" ]
+  },
+  {
+    "id": "sword_forged",
```
Comment (kevingranade) [inline data/json/items/melee.json]: This seems redundant with the arming sword, what differentiates them?

File: data/json/items/ranged.json [hunk 3]
```diff
@@ -576,5 +576,32 @@
     "clip_size": 100,
     "reload": 0,
     "valid_mod_locations": [ [ "sling", 1 ] ]
+  },
+  {
+    "id": "fire_lance",
```
Comment (kevingranade) [inline data/json/items/ranged.json]: This is utter nonsense.  
One, the difficulty to craft is ridiculously low.  
Two, the damage on this thing is crazy high.  
Three, the original is explicitly single-shot.  
The source weapon is a glorified firecracker on a stick, not a serious projectile weapon.

Leave it in MST and if you want a craftable early firearm look into casting a Gonne.
If you want a fire lance, look into making it some kind of single use distracting/stunning thing that deals minor damage.
You might say, "but that makes it useless", which is correct, they ARE useless for most individual purposes.

File: data/json/recipes/recipe_others.json [hunk 4]
```diff
@@ -4389,5 +4389,33 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_MATERIALS",
+    "skill_used": "fabrication",
+    "difficulty": 2,
+    "time": 10000,
+    "batch_time_factors": [ 90, 4 ],
+    "autolearn": true,
+    "using": [ [ "forging_standard", 1 ] ],
+    "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ] ] ],
+    "components": [ [ [ "tin", 10 ], [ "solder_wire", 10 ] ], [ [ "copper", 50 ] ] ]
```
Comment (kevingranade) [inline data/json/recipes/recipe_others.json]: Solder wire is not a reliable source of tin for this purpose.

File: data/json/snippets.json [hunk 5]
```diff
@@ -1791,6 +1791,39 @@
     "category": "note",
     "text": [ "\"Husband 30 yrs died in bombs.  Mushrooms brought him back.  GIVE YOURSELF TO THEm AND THEY WILL HELP YOU\"" ]
   },
+  {
```
Comment (kevingranade) [inline data/json/snippets.json]: Cut all these snippets, none of them make sense to appear in the dda game setting.

File: data/json/items/ranged.json (hunk 3 above)
Comment (Mooses2k) [reply data/json/items/ranged.json]: I have no dog in this race. I nerfed it to oblivion.

File: data/json/items/melee.json (hunk 2 above)
Comment (Mooses2k) [reply data/json/items/melee.json]: Material of make, but eh, it's out of place, you're right. I'll get rid of it. Replacing with xiphos where appropriate.

**Comment by Mooses2k** [comment]
Balancing remaining weapons now.

File: data/json/recipes/recipe_others.json [hunk 6]
```diff
@@ -4389,5 +4389,33 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_MATERIALS",
+    "skill_used": "fabrication",
+    "difficulty": 2,
+    "time": 10000,
+    "batch_time_factors": [ 90, 4 ],
+    "autolearn": true,
+    "using": [ [ "forging_standard", 1 ] ],
+    "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ] ] ],
+    "components": [ [ [ "tin", 12 ], [ [ "copper", 50 ] ] ]
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_others.json]:
```suggestion
    "components": [ [ [ "tin", 12 ], [ "copper", 50 ] ] ]
```

**Comment by Mooses2k** [comment]
Rebalancing done based on size of weapon, making similar to existing spear, one-handed, and two-handed swords.

**Comment by Mooses2k** [comment]
```
12:19:36 CCACHE_CPP2=1 ccache clang++ -ffast-math -Os  -Werror -Wall -Wextra -Woverloaded-virtual -Wpedantic -Wmissing-declarations    -fsigned-char  -std=c++14 -MMD -MP -m32 -D_GNU_SOURCE -I/usr/include/ncursesw -Itools/format -Isrc tools/format/format.cpp src/json.cpp -o tools/format/json_formatter.cgi
12:19:36 Makefile:997: recipe for target 'astyle-check' failed
12:19:36 make: *** [astyle-check] Error 1
```
This doesn't look like my fault. Rebuild?

**Comment by SirPendrak** [comment]
Like.... what the hell was wrong with estoc, longsword, arming sword? Estoc is now like 40% worse version of rapier. Arming sword is now even more shitty than broadsword. Longsword is now usless, pick katana or nodachi. Werent you supposed to mostly change just new stuff, without trashing some of existing ones without explaination, while leaving many similar weapons unchanged?

**Comment by Mooses2k** [comment]
Just dialing it in. I'll have another go at adjustment. Thanks for feedback. :)

**Comment by kevingranade** [comment]
Unnecessary scatology aside, I also didn't expect you to change balance of existing weapons, just fit the new ones into the existing continuum.
If a larger rebalance needs to happen, it needs to happen in a different PR with some significant discussion.

File: data/json/items/book/martial.json [hunk 7]
```diff
@@ -214,5 +214,13 @@
...
     "name_plural": "The Way of the Spear",
     "description": "A complete guide to Sōjutsu."
+  },
+  {
+    "id": "manual_swordsmanship",
+    "copy-from": "book_martial",
+    "type": "GENERIC",
+    "name": "Historic European Swordfighting",
+    "name_plural": "Historic European Swordfighting",
+    "description": "A complete guide to Medieval Swordsmanship.  Compares the German and Italian traditions for longsword and side sword, in and out of armor, with and without shield."
```
Comment (LaVeyanFiend) [inline data/json/items/book/martial.json]:
```suggestion
    "description": "A complete guide to medieval swordsmanship.  Compares the German and Italian traditions for longsword and side sword, in and out of armor, with and without shield."
```

File: data/json/martialarts.json [hunk 8]
```diff
@@ -969,5 +969,65 @@
       "spear_survivor",
       "spear_wood"
     ]
+  },
+  {
+    "type": "martial_art",
+    "id": "style_swordsmanship",
+    "name": "Medieval Swordsmanship",
+    "description": "The art of the longsword and sword & shield, preceding the later development of fencing.  Designed for combat both unarmored and in armor, it includes grappling as well as defensive and offensive sword techniques.  This treatise compares the Italian and German traditions of medieval combat with detailed step-by-step pictures.",
```
Comment (LaVeyanFiend) [inline data/json/martialarts.json]:
```suggestion
    "description": "The art of the longsword and sword & shield, preceding the later development of fencing.  Designed for combat both unarmored and in armor, it includes grappling as well as defensive and offensive sword techniques.",
```

Comment (LaVeyanFiend) [reply data/json/martialarts.json]: Text from the manual leaked into the MA selection menu description, I see.

File: data/json/itemgroups/mansion.json [hunk 9]
```diff
@@ -400,7 +419,7 @@
       [ "pike_fake", 50 ],
       { "group": "broadsword_fake+scabbard", "prob": 17 },
       [ "mace_fake", 6 ],
-      [ "morningstar_fake", 5 ]
+      [ "morningstar_fake", 5 ],
```
Comment (LaVeyanFiend) [inline data/json/itemgroups/mansion.json]:
```suggestion
      [ "morningstar_fake", 5 ]
```

Comment (LaVeyanFiend) [reply data/json/itemgroups/mansion.json]: Shoo, extra seperator, shoo!

File: data/json/martialarts.json (hunk 8 above)
Comment (LaVeyanFiend) [reply data/json/martialarts.json]: There are no pictures in the martial arts selection menu. 😛

File: data/json/itemgroups/mansion.json (hunk 9 above)
Comment (Mooses2k) [reply data/json/itemgroups/mansion.json]: Nope, thx. Just moving fast and breaking my own things.

File: data/json/recipes/recipe_others.json [hunk 10]
```diff
@@ -4389,5 +4389,33 @@
...
   }
 ]
```
Comment (LaVeyanFiend) [inline data/json/recipes/recipe_others.json]:
```suggestion
]

```

Comment (LaVeyanFiend) [reply data/json/recipes/recipe_others.json]: No newline makes game mad.

File: data/json/recipes/recipe_others.json [hunk 11]
```diff
@@ -4389,5 +4389,33 @@
...
+    "subcategory": "CSC_OTHER_MATERIALS",
+    "skill_used": "fabrication",
+    "difficulty": 2,
+    "time": 10000,
+    "batch_time_factors": [ 90, 4 ],
+    "autolearn": true,
+    "using": [ [ "forging_standard", 1 ] ],
+    "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ] ] ],
+    "components": [ [ [ "tin", 12 ], [ [ "copper", 50 ] ] ]
   }
```
Comment (LaVeyanFiend) [inline data/json/recipes/recipe_others.json]: Gorgon not liking this red endbracket
image

File: data/json/itemgroups/books.json [hunk 12]
```diff
@@ -49,10 +49,11 @@
     "items": [
       [ "manual_eskrima", 100 ],
       [ "manual_krav_maga", 100 ],
-      [ "manual_ninjutsu", 100 ],
+      [ "manual_ninjutsu", 50 ],
```
Comment (Hymore246) [inline data/json/itemgroups/books.json]: Any reason why this was changed from 100 to 50?

Comment (Mooses2k) [reply data/json/itemgroups/books.json]: Out of scope to this PR, sure, but come on. The relative frequency versus the other manuals should be lower to reflect reality.

**Comment by Mooses2k** [comment]
Could I get help figuring out how to fix this test which is failing

File: data/json/item_groups.json [hunk 13]
```diff
@@ -8386,7 +8393,6 @@
     ]
   },
   {
-    "id": "museum_primitive",
```
Comment (Night-Pryanik) [inline data/json/item_groups.json]: Why are you removing `id` of this item group?

Comment (Mooses2k) [reply data/json/item_groups.json]: Thanks, whoops. Moved all museum stuff to it's own json as part of overall json organization. This was just a screwup - didn't get the whole category moved. Good catch.

File: data/json/itemgroups/books.json (hunk 12 above)
Comment (kevingranade) [reply data/json/itemgroups/books.json]: Sneaking in out of scope things in your PRs is a great way to have them stall forever.

**Comment by Mooses2k** [comment]
[Table of Medieval weapons inclusion.pdf](https://github.com/CleverRaven/Cataclysm-DDA/files/3283435/Table.of.Medieval.weapons.inclusion.pdf)

File: data/json/techniques.json [hunk 14]
```diff
@@ -1073,6 +1073,7 @@
   },
   {
     "type": "technique",
+<<<<<<< master
```
Comment (kevingranade) [inline data/json/techniques.json]:
```suggestion
```

**Comment by Mooses2k** [comment]
Jenkins rebuild

File: data/json/items/ammo.json [hunk 15]
```diff
@@ -1372,5 +1372,22 @@
     "count": 7,
     "effects": [ "NEVER_MISFIRES" ],
     "use_action": "RPGDIE"
+},
```
Comment (kevingranade) [inline data/json/items/ammo.json]:
```suggestion
  },
```

File: data/json/items/ammo.json [hunk 16]
```diff
@@ -1372,5 +1372,22 @@
...
+{
```
Comment (kevingranade) [inline data/json/items/ammo.json]:
```suggestion
  {
```

File: data/json/martialarts.json [hunk 17]
```diff
@@ -1009,5 +1009,64 @@
       "tec_wingchun_disarm"
     ],
     "weapons": [ "butterfly_swords", "long_pole" ]
+},
```
Comment (kevingranade) [inline data/json/martialarts.json]:
```suggestion
  },
```

File: data/json/martialarts.json [hunk 18]
```diff
@@ -1009,5 +1009,64 @@
...
+{
```
Comment (kevingranade) [inline data/json/martialarts.json]:
```suggestion
  {
```

File: data/json/techniques.json [hunk 19]
```diff
@@ -1134,5 +1134,67 @@
     "unarmed_allowed": true,
     "disarms": true,
     "messages": [ "You disarm %s", "<npcname> disarms %s" ]
+},
```
Comment (kevingranade) [inline data/json/techniques.json]:
```suggestion
  },
```

File: data/json/techniques.json [hunk 20]
```diff
@@ -1134,5 +1134,67 @@
...
+{
```
Comment (kevingranade) [inline data/json/techniques.json]:
```suggestion
  {
```

**Comment by kevingranade** [comment]
My suggestions fix some merge errors, but even after they're applied there are load errors remaining that I haven't found the source of.

**Comment by AMurkin** [comment]
I'm able to load a new world with this changes:
```diff
diff --git a/data/json/items/ammo.json b/data/json/items/ammo.json
index 45d6824793..ef15a71234 100644
--- a/data/json/items/ammo.json
+++ b/data/json/items/ammo.json
@@ -1372,8 +1372,8 @@
     "count": 7,
     "effects": [ "NEVER_MISFIRES" ],
     "use_action": "RPGDIE"
-},
-{
+  },
+  {
     "id": "scrap_bronze",
     "type": "AMMO",
     "category": "spare_parts",
diff --git a/data/json/martialarts.json b/data/json/martialarts.json
index 44e8c78d9c..e86f8b2766 100644
--- a/data/json/martialarts.json
+++ b/data/json/martialarts.json
@@ -1009,8 +1009,9 @@
       "tec_wingchun_disarm"
     ],
     "weapons": [ "butterfly_swords", "long_pole" ]
-},
-{
+  },
+  {
+    "type": "martial_art",
     "id": "style_swordsmanship",
     "name": "Medieval Swordsmanship",
     "description": "The art of the longsword and sword & shield, preceding the later development of fencing.  Designed for combat both unarmored and in armor, it includes grappling as well as defensive and offensive sword techniques.  This treatise compares the Italian and German traditions of medieval combat with detailed step-by-step pictures.",
diff --git a/data/json/professions.json b/data/json/professions.json
index 061729b2e5..201c3764df 100644
--- a/data/json/professions.json
+++ b/data/json/professions.json
@@ -3752,6 +3752,7 @@
     }
   },
   {
+    "type": "profession",
     "ident": "politician",
     "name": "Career Politician",
     "description": "You've spent your life appealing to the people, persuading many and promising much throughout your time in office.  Now that your voting base wants to eat you alive, winning hearts and minds just got that much harder.",
@@ -3802,4 +3803,4 @@
      "female": [ "bra", "panties" ] 
     }
   }
-]
\ No newline at end of file
+]
diff --git a/data/json/techniques.json b/data/json/techniques.json
index 48827abc4d..6944ffb3b2 100644
--- a/data/json/techniques.json
+++ b/data/json/techniques.json
@@ -1134,8 +1134,9 @@
     "unarmed_allowed": true,
     "disarms": true,
     "messages": [ "You disarm %s", "<npcname> disarms %s" ]
-},
-{
+  },
+  {
+    "type": "technique",
     "id": "tec_swordsmanship_grab",
     "name": "grab and pommel strike",
     "min_melee": 3,
diff --git a/data/mods/Medieval_Stuff/item_groups.json b/data/mods/Medieval_Stuff/item_groups.json
index 526e5179c5..c1b08ddc94 100644
--- a/data/mods/Medieval_Stuff/item_groups.json
+++ b/data/mods/Medieval_Stuff/item_groups.json
@@ -34,7 +34,7 @@
   {
     "id": "museum_melee",
     "type": "item_group",
-    "items": [ [ "ji", 2 ], [ "dao", 3 ], [ "sword_xiphos", 4 ], [ "sword_forged", 3 ], [ "khopesh", 4 ], [ "spear_dory", 3 ] ]
+    "items": [ [ "ji", 2 ], [ "dao", 3 ], [ "sword_xiphos", 4 ], [ "khopesh", 4 ], [ "spear_dory", 3 ] ]
   },
   {
     "id": "museum_guns",
diff --git a/data/mods/Medieval_Stuff/professions.json b/data/mods/Medieval_Stuff/professions.json
index 5fb12ddbec..e8d34144a6 100644
--- a/data/mods/Medieval_Stuff/professions.json
+++ b/data/mods/Medieval_Stuff/professions.json
@@ -78,7 +78,7 @@
           "waterskin"
         ],
         "entries": [
-          { "item": "xiphos", "container-item": "baldric" },
+          { "item": "sword_xiphos", "container-item": "baldric" },
           { "item": "knife_hunting", "container-item": "sheath" },
           { "item": "javelin_iron", "count": 2 }
         ]
```

Not sure how data/json/professions.json (Career Politician) has changed in this PR.

**Comment by Hymore246** [comment]
If the new death blow technique is a mordhau, why does it increase cut and stab damage? A  [real mordhau](https://en.wikipedia.org/wiki/Mordhau_(weaponry)) is flipping your sword around and bashing your opponent with pommel or crossguard. The tech should reduce the cut and stab damage of the attack to 0.

**Comment by LaVeyanFiend** [comment]
> If the new death blow technique is a mordhau, why does it increase cut and stab damage? A [real mordhau](https://en.wikipedia.org/wiki/Mordhau_(weaponry)) is flipping your sword around and bashing your opponent with pommel or crossguard. The tech should reduce the cut and stab damage of the attack to 0.

Unless there's a way to convert damage types inside a technique, swords would deal 0 bash damage since the majority of them are cut/pierce only.

File: data/json/items/tools.json [hunk 21]
```diff
@@ -9160,6 +9160,82 @@
...
+    "category": "weapons",
+    "name": "dao",
+    "name_plural": "dao",
+    "description": "An ancient Chinese sword consisting of a curved blade and a guard with a cupped design.  Existing since the Shang dynasty, this one is made of bronze.  One of the four major weapons in folklore, alongside the jian sword, qiang spear, and gun staff.",
+    "weight": 850,
+    "volume": 7,
+    "price": 12500,
+    "to_hit": 2,
+    "bashing": 6,
+    "cutting": 24,
```
Comment (kevingranade) [inline data/json/items/tools.json]: A bit out of line for a small bronze sword.
```suggestion
    "cutting": 18,
```

**Pull request merged**